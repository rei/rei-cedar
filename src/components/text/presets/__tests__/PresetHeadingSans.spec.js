import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import PresetHeadingSans from '../PresetHeadingSans.vue';


describe('PresetHeadingSans', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(PresetHeadingSans, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(PresetHeadingSans, {
        props: { scale: '1' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
