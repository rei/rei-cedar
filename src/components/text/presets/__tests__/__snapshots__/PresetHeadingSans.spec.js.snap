// Vitest Snapshot v1

exports[`PresetHeadingSans > default configuration > renders correctly 1`] = `
<p
  class="cdr-text cdr-heading-sans"
  style="--cdr-heading-sans-font-size: var(--cdr-type-scale-3); --cdr-heading-sans-line-height: calc(var(--cdr-type-scale-3) * var(--cdr-heading-sans-line-height-ratio));"
>
  <!-- @slot CdrText content -->
  
  
  paragraph
  
  
</p>
`;

exports[`PresetHeadingSans > passing a scale > renders correctly 1`] = `
<p
  class="cdr-text cdr-heading-sans"
  style="--cdr-heading-sans-font-size: var(--cdr-type-scale-1); --cdr-heading-sans-line-height: calc(var(--cdr-type-scale-1) * var(--cdr-heading-sans-line-height-ratio));"
>
  <!-- @slot CdrText content -->
  
  
  paragraph
  
  
</p>
`;
